src/mongo_logger.ts

```diff
@@ -383,7 +413,8 @@ export function stringifyWithMaxLen(
   maxDocumentLength: number,
   options: EJSONOptions = {}
 ): string {
-  let strToTruncate: string;
+  let strToTruncate = '';
```

Comment:
Non-blocking and definitely only tangentially related: the `.slice()` call below can end up truncating `strToTruncate` in the middle of a character. That won't happen a lot if the string mostly just contains ASCII data, but since this is a very low-level generic utility, it might be worth accounting for that case (e.g. `if (strToTruncate.length > maxDocumentLength && strToTruncate.charCodeAt(maxDocumentLength-1) !== strToTruncate.codePointAt(maxDocumentLength-1)) maxDocumentLength--;`).

Reply:
This is something we will solve in [NODE-5839](https://jira.mongodb.org/browse/NODE-5839 ).